test/compress/destructuring.js

```diff
@@ -200,3 +200,13 @@ destructuring_remove_unused_2: {
         }
     }
 }
+
+object_destructuring_may_need_parentheses: {
+    beautify = {
+        ecma: 6
+    }
+    input: {
+        ({a, b} = {a: 1, b: 2});
```

Comment:
This doesn't seem to be valid lhs expression, but since we post-process the ast, there isn't a trivial way to detect this.